azavea/loam
# Add basic support for vector datasets

## Overview

Allows opening vector datasets via `loam.open()`. Also includes wrappers for
- `GDALDatasetGetLayerCount()` (for determining if a dataset has vector layers)
- `GDALVectorTranslate()` (API equivalent of ogr2ogr command)
- Updates the demo to support multi-file formats such as Shapefiles

### Demo

![Screenshot from 2022-11-07 14-08-50](https://user-images.githubusercontent.com/447977/200394074-9a8fc534-860b-45fb-a568-5129ce7a1389.png)

### Notes

- There are lots of different vector formats, so I'm not certain that this works with all of them, but it worked with GeoJSON and Shapefiles, which are the two that we most commonly encounter.

## Testing Instructions

 * Review the code and tests and confirm that the new tests sufficiently cover the added functionality.
 * Review the README and make sure it accurately describes the changed and added functionality.
 * CI should pass
 * If you want, play around with the demo files to add other functionality and confirm that it works as expected.

## Checklist


Resolves #101, Resolves #82
